src/lib/kube-enclave.ts

```diff
@@ -1,58 +1,62 @@
-import { k8sApiCall } from './api'
+import { k8sApiCall, toaUpdateJobStatus } from './api'
 import { Enclave, IEnclave } from './enclave'
 import { createKubernetesJob, filterDeployments } from './kube'
 import {
     KubernetesApiJobsResponse,
     KubernetesApiResponse,
     KubernetesJob,
+    KubernetesPod,
     ManagementAppGetReadyStudiesResponse,
     ManagementAppJob,
     TOAGetJobsResponse,
 } from './types'
 
-class KubernetesEnclave extends Enclave<KubernetesApiJobsResponse> implements IEnclave<KubernetesApiJobsResponse> {
+class KubernetesEnclave extends Enclave<KubernetesJob> implements IEnclave<KubernetesJob> {
     filterJobsInEnclave(
         bmaReadysResults: ManagementAppGetReadyStudiesResponse,
         toaGetJobsResult: TOAGetJobsResponse,
-        runningJobsInEnclave: KubernetesApiJobsResponse[],
+        runningJobsInEnclave: KubernetesJob[],
     ): ManagementAppGetReadyStudiesResponse {
         console.log('Filtering Kubernetes jobs...')
-        /* v8 ignore next */
-        if (!runningJobsInEnclave?.length) return bmaReadysResults || { jobs: [] }
+        const jobs: ManagementAppJob[] = []
+        const jobsInEnclaveIds = runningJobsInEnclave
+            ?.filter(
+                (job) =>
+                    job.metadata?.labels?.['managed-by'] === 'setup-app' &&
```

Comment:
Maybe hoist `setup-app` and `research-container` as constants since they are used in multiple places?